packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,230 @@
+    padding: 25px 12px;
+`;
+
+const Text = styled.div<{ isPixelType: boolean; areChunksUsed?: boolean }>`
+    font-family: ${props => (props.isPixelType ? 'PixelOperatorMono8' : 'RobotoMono')};
+    font-size: ${props => (props.isPixelType ? '12' : '18')}px;
+    color: white;
+    word-break: break-word;
+    max-width: ${props => (props.areChunksUsed ? '100%' : '216px')};
+`;
+
+const Row = styled.div<{ isPixelType: boolean; isAlignedRight?: boolean }>`
+    display: flex;
+    align-items: center;
+    gap: ${props => (props.isPixelType ? '8' : '9')}px;
+    justify-content: ${props => (props.isAlignedRight ? 'flex-end' : 'flex-start')};
+`;
+
+const Chunks = styled.div<{ isPixelType: boolean }>`
+    display: flex;
+    flex-direction: column;
+    gap: ${props => (props.isPixelType ? '6' : '5')}px;
+`;
+
+const Wrapper = styled.div`
+    position: relative;
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+`;
+
+const Divider = styled.div<{ isPixelType: boolean }>`
+    width: 100%;
+    height: 1px;
+    background: #2b2b2b;
+    margin: ${props => (props.isPixelType ? '10px' : '0')} 0;
+`;
+
+const AddressLabel = styled.span<{ isPixelType: boolean }>`
+    font-weight: 600;
+    color: #868686;
+    font-size: ${variables.FONT_SIZE.TINY};
+    text-transform: uppercase;
+    position: absolute;
+    background: #000000;
+    padding: 0 10px;
+    text-align: center;
+    left: 50%;
+    bottom: ${props => (props.isPixelType ? '2' : '-7')}px;
+    transform: translate(-50%, 0);
+`;
+
+const StyledNextIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    bottom: ${props => (props.isPixelType ? '13' : '20')}px;
+    right: ${props => (props.isPixelType ? '-95' : '35')}px;
+`;
+
+const StyledContinuesIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    top: ${props => (props.isPixelType ? '10' : '25')}px;
+    right: ${props => (props.isPixelType ? '82' : '97')}px;
+`;
+
+const StyledChunkedNextIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    bottom: ${props => (props.isPixelType ? '13' : '20')}px;
+    right: -67px;
+`;
+
+const StyledChunkedContinuesIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    top: ${props => (props.isPixelType ? '10' : '25')}px;
+    right: 68px;
+`;
+
+export interface DeviceDisplayProps {
+    address: string;
+    network?: string;
+}
+
+export const DeviceDisplay = ({ address, network }: DeviceDisplayProps) => {
+    const selectedDeviceInternalModel = useSelector(selectDeviceInternalModel);
+    const unavailableCapabilities = useSelector(selectDeviceUnavailableCapabilities);
+    const addressDisplay = useSelector(selectAddressDisplay);
+
+    const areChunksUsed =
+        addressDisplay === AddressDisplayOptions.CHUNKED && !unavailableCapabilities?.chunkify;
+    const isPixelType = selectedDeviceInternalModel !== DeviceModelInternal.T2T1;
+    const isPaginated = network === 'cardano';
```

Comment:
Theoretically it makes more sense to do it based on address length (but it adds complexity like chunked/original and the device type). In reality it can only happen on Cardano (currently and also in foreseeable future).